app/controllers/actions/GroupAction.scala

```diff
@@ -87,6 +87,21 @@ class GroupAction @Inject()
     }
   }
 
+  def withAccessGroup(groupId: GroupId, isCustom: Boolean = true)
+                     (callback: AccessGroup => Future[Result])
+                     (implicit ec: ExecutionContext, hc: HeaderCarrier,
+                      request: MessagesRequest[AnyContent], appConfig: AppConfig): Future[Result] = {
+    if (isCustom) {
+      groupService
+        .getGroup(groupId)
+        .flatMap(_.fold(groupNotFound())(customGroup => callback(customGroup)))
+    } else {
+      taxGroupService
+        .getGroup(groupId)
+        .flatMap(_.fold(groupNotFound())(taxGroup => callback(taxGroup)))
+    }
+  }
+
```

Comment:
Not sure how I feel about this, since it's basically the same as `withAccessGroupForAuthorisedOptedAgent` but without the ARN which you can ignore. I guess the other difference is it doesn't go through auth... so it doesn't seem like a good idea

Reply:
so what do u suggest?